File: THFoundation/THFoundation.m
// THFoundation.m

#import "THFoundation.h"

//--------------------------------------------------------------------------------------------------------------------------------------------
CGFloat CGFloatFloor(CGFloat value) { return floor(value); } 	/* arrondi inferieur */
CGFloat CGFloatCeil(CGFloat value) { return ceil(value); }		/* arrondi supperieur */
CGFloat CGFloatRint(CGFloat value) { return rint(value); }		/* arrondi normal */
//--------------------------------------------------------------------------------------------------------------------------------------------


//--------------------------------------------------------------------------------------------------------------------------------------------
BOOL TH_IsEqualNSString(NSString *string, NSString *anotherString)
{
	if (string==anotherString)
		return YES;
	if (string!=nil && anotherString!=nil && CFStringCompare((__bridge CFStringRef)string,(__bridge CFStringRef)anotherString,0)==kCFCompareEqualTo)
		return YES;
	return NO;
}

BOOL TH_IsEqualNSPoint(NSPoint point, NSPoint anotherPoint, CGFloat tolerance)
{
	if (tolerance==0.0)
		return (point.x==anotherPoint.x && point.y==anotherPoint.y)?YES:NO;
	CGFloat dX=point.x-anotherPoint.x;
	CGFloat dY=point.y-anotherPoint.y;
	return ((dX<=tolerance && dX>=tolerance*-1.0) && (dY<=tolerance && dY>=tolerance*-1.0))?YES:NO;
}
//--------------------------------------------------------------------------------------------------------------------------------------------


//--------------------------------------------------------------------------------------------------------------------------------------------
NSColor* TH_RGBACOLOR(CGFloat r, CGFloat g, CGFloat b, CGFloat a)
{
	return [NSColor colorWithDeviceRed:r/255.0 green:g/255.0 blue:b/255.0 alpha:a];
}

NSColor* TH_RGBCOLOR(CGFloat r, CGFloat g, CGFloat b)
{
	return [NSColor colorWithDeviceRed:r/255.0 green:g/255.0 blue:b/255.0 alpha:1.0];
}
//--------------------------------------------------------------------------------------------------------------------------------------------

File: THFoundation/THFoundation.swift
// THFoundation.swift

#if os(macOS)
	import Cocoa
#elseif os(iOS)
	import UIKit
#endif

//-----------------------------------------------------------------------------------------------------------------------------------------
#if os(macOS)
typealias TH_NSUI_Image = NSImage
#elseif os(iOS)
typealias TH_NSUI_Image = UIImage
#endif
//-----------------------------------------------------------------------------------------------------------------------------------------


//--------------------------------------------------------------------------------------------------------------------------------------------
//func CGFloatFloor(_ value: CGFloat) -> CGFloat { value.rounded(.down) } 	// arrondi inferieur */
//CGFloat CGFloatCeil(CGFloat value) { return ceil(value); }		/* arrondi supperieur */
//CGFloat CGFloatRint(CGFloat value) { return rint(value); }		/* arrondi normal */
//--------------------------------------------------------------------------------------------------------------------------------------------


//-----------------------------------------------------------------------------------------------------------------------------------------
func THLocalizedString(_ string: String) -> String {
	NSLocalizedString(string, comment: "")
}

func THLocalizedStringClass(_ classObject: NSObject, _ string: String) -> String {
	NSLocalizedString(string, tableName: classObject.th_className, comment: "")
}

func THLocalizedStringTable(_ table: String, _ string: String) -> String {
	NSLocalizedString(string, tableName: table, comment: "")
}
//-----------------------------------------------------------------------------------------------------------------------------------------


//-----------------------------------------------------------------------------------------------------------------------------------------
#if os(macOS)
func TH_RGBCOLOR(_ red: CGFloat, _ green: CGFloat, _ blue: CGFloat, _ alpha: CGFloat = 1.0) -> NSColor {
	NSColor(deviceRed: red / 255.0, green: green / 255.0, blue: blue / 255.0, alpha: alpha)
}
#elseif os(iOS)
func TH_RGBCOLOR(_ red: CGFloat, _ green: CGFloat, _ blue: CGFloat, _ alpha: CGFloat = 1.0) -> UIColor {
	UIColor(red: red / 255.0, green: green / 255.0, blue: blue / 255.0, alpha: alpha)
}
#endif
//-----------------------------------------------------------------------------------------------------------------------------------------
